docroot/.htaccess

```diff
@@ -69,6 +69,20 @@ AddEncoding gzip svgz
 <IfModule mod_rewrite.c>
   RewriteEngine on
 
+  # Forbid access to install and update files.
+  RewriteRule ^(core/install|update)\.php - [F]
+
+  # Forbid access to admin, node, and user paths on domains other than admin.foia.gov and local.dojfoia.gov.
+  RewriteCond %{ENV:AH_SITE_ENVIRONMENT} ^(prod|test|dev)$
+  RewriteCond %{HTTP_HOST} !(.*)admin\.foia\.gov$ [NC]
+  RewriteRule ^((admin|node|user)($|/)) - [F]
+
+  # Redirect all HTTP server traffic to HTTPS.
+  RewriteCond %{ENV:AH_SITE_ENVIRONMENT} ^(prod|test|dev)$
+  RewriteCond %{HTTPS} off
+  RewriteCond %{HTTP:X-Forwarded-Proto} !https
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [NE,L,R=301]
```

Comment:
What does the `NE` flag do?

Reply:
No Escape - doesn't convert special characters to hexcode (e.g. & => %26) potentially resulting in double encoding or 404 not found error conditions.